satellite/console/service.go

```diff
@@ -141,6 +119,41 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		return nil, errs.New(internalErrMsg)
 	}
 
+	// TODO: remove after vanguard release
+	// when user uses an open source partner referral link, there won't be a registration token in the link.
+	// therefore, we need to create one so we can still control the project limit on the account level
+	var registrationToken *RegistrationToken
+	if user.PartnerID != "" {
+		registrationToken, err = s.store.RegistrationTokens().Create(ctx, 1)
```

Comment:
do we really want to limit them with 1 project? Is it a requirenment or just a typo?